### PR TITLE
Fix 9365: Add gn check to npm scripts

### PR DESCRIPTION
**Pull request by jumde**
Resolves https://github.com/brave/brave-browser/issues/9365

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: DEPS [hunk 1]
```diff
@@ -13,7 +13,7 @@ deps = {
   "vendor/bip39wally-core-native": "https://github.com/brave-intl/bip39wally-core-native.git@13bb40a215248cfbdd87d0a6b425c8397402e9e6",
   "vendor/bat-native-anonize": "https://github.com/brave-intl/bat-native-anonize.git@e3742ba3e8942eea9e4755d91532491871bd3116",
   "vendor/bat-native-tweetnacl": "https://github.com/brave-intl/bat-native-tweetnacl.git@800f9d40b7409239ff192e0be634764e747c7a75",
-  "vendor/bat-native-usermodel": "https://github.com/brave-intl/bat-native-usermodel.git@865ba342737c09b13ee18e45b8ece5970bb77246",
+  "vendor/bat-native-usermodel": "https://github.com/brave-intl/bat-native-usermodel.git@02b8c81c94072c67fe00108feb90786e088d4d26",
```
Comment (bridiver) [inline DEPS:16]: ?

File: vendor/brave-ios/BUILD.gn [hunk 2]
```diff
@@ -69,6 +69,7 @@ ios_framework_bundle("brave_rewards_ios_framework") {
   deps = [
     "//brave/vendor/bat-native-ledger",
     "//brave/vendor/bat-native-ads",
+    "//net:net",
```
Comment (bridiver) [inline vendor/brave-ios/BUILD.gn:72]: ?

File: components/binance/browser/BUILD.gn [hunk 3]
```diff
@@ -20,8 +20,12 @@ source_set("browser") {
 
   deps = [
     "//base",
+    "//brave/common:pref_names",
```
Comment (bridiver) [inline components/binance/browser/BUILD.gn:23]: if these are missing deps you discovered when running it, can we at least put them in a separate commit?

File: DEPS (hunk 1 above)
Comment (jumde) [reply DEPS:16]: Deps were fixed in `bat-native-usermodel`: https://github.com/brave-intl/bat-native-usermodel/issues/31 - I'll move this to a separate commit.

File: components/binance/browser/BUILD.gn (hunk 3 above)
Comment (jumde) [reply components/binance/browser/BUILD.gn:23]: moved to a separate commit.

File: vendor/brave-ios/BUILD.gn (hunk 2 above)
Comment (jumde) [reply vendor/brave-ios/BUILD.gn:72]: For ios:

```
09:07:40  > brave@1.13.21 gn_check /Users/jenkins/jenkins/workspace/pr-brave-browser-gn_check
09:07:40  > node ./scripts/commands.js gn_check "Release" "--target_os=ios"
09:07:40  
09:07:40  Recursing through GRD to find GRDP files...
09:07:40  Done recursing through GRD to find GRDP files.
09:07:40  using git cache path /Users/jenkins/cache
09:07:40  -------------------------------
09:07:40  /Users/jenkins/jenkins/workspace/pr-brave-browser-gn_check/src
09:07:40  > gn check /Users/jenkins/jenkins/workspace/pr-brave-browser-gn_check/src/out/ios_Release //brave/*
09:07:44  ERROR at //brave/vendor/brave-ios/Ledger/BATBraveLedger.mm:31:10: Can't include this header from here.
09:07:44  #import "net/base/registry_controlled_domains/registry_controlled_domain.h"
09:07:44           ^----------------------------------------------------------------
09:07:44  The target:
09:07:44    //brave/vendor/brave-ios:brave_rewards_ios_framework_arch_shared_library_sources
09:07:44  is including a file from the target:
09:07:44    //net:net
09:07:44  
```

File: components/brave_wallet/browser/BUILD.gn [hunk 4]
```diff
@@ -16,6 +16,8 @@ source_set("browser") {
 
     deps = [
         "//base",
+        "//brave/common:common",
```
Comment (iefremov) [inline components/brave_wallet/browser/BUILD.gn:19]: we want to avoid this, don't we?

File: components/brave_prochlo/BUILD.gn [hunk 5]
```diff
@@ -13,6 +13,7 @@ source_set("brave_prochlo") {
 
   public_deps = [
     ":prochlo_proto",
+    "//base:base",
```
Comment (iefremov) [inline components/brave_prochlo/BUILD.gn:16]: do we really need adding `base`?

Comment (jumde) [reply components/brave_prochlo/BUILD.gn:16]: @iefremov - I get this error if we don't:

```
> gn check src/out/Component //brave/*
ERROR at //brave/components/brave_prochlo/brave_prochlo_message.h:11:11: Can't include this header from here.
#include "base/time/time.h"
          ^---------------
The target:
  //brave/components/brave_prochlo:brave_prochlo
is including a file from the target:
  //base:base

ERROR at //brave/components/brave_prochlo/brave_prochlo_crypto.h:11:11: Can't include this header from here.
#include "base/macros.h"
          ^------------
The target:
  //brave/components/brave_prochlo:brave_prochlo
is including a file from the target:
  //base:base
```

File: components/brave_wallet/browser/BUILD.gn (hunk 4 above)
Comment (jumde) [reply components/brave_wallet/browser/BUILD.gn:19]: @iefremov - error if this is missing

```
> gn check /src/out/Component //brave/*
ERROR at //brave/components/brave_wallet/browser/brave_wallet_service.cc:17:11: Include not allowed.
#include "brave/common/extensions/extension_constants.h"
          ^--------------------------------------------
It is not in any dependency of
  //brave/components/brave_wallet/browser:browser
The include file is in the target(s):
  //brave/common:common
which should somehow be reachable.
```

File: Jenkinsfile [hunk 6]
```diff
@@ -141,7 +141,7 @@ def startBraveBrowserBuild() {
                                 credentials('brave-builds-github-token-for-pr-builder')
                                 github('brave/devops', 'https')
                             }
-                            branch('master')
+                            branch('gn_check')
```
Comment (mihaiplesa) [inline Jenkinsfile:144]: this needs revert

File: vendor/brave-ios/BUILD.gn (hunk 2 above)
Comment (kylehickinson) [reply vendor/brave-ios/BUILD.gn:72]: Yeah we were piggybacking on ledger/ads usage rather than import it ourself

File: DEPS (hunk 1 above)
Comment (bridiver) [reply DEPS:16]: we should move this into brave-core, or at the very least move the gn build config into brave-core

File: Jenkinsfile (hunk 6 above)
Comment (jumde) [reply Jenkinsfile:144]: @mihaiplesa - will revert once the PR is ready to merge.

File: vendor/brave-ios/BUILD.gn (hunk 2 above)
Comment (bridiver) [reply vendor/brave-ios/BUILD.gn:72]: @kylehickinson @tmancey it would be better if that was handled inside ledger now that we allow chromium deps there, but sounds like this is needed for now

File: DEPS (hunk 1 above)
Comment (jumde) [reply DEPS:16]: cc: @tmancey @dimmu 

Filed an issue: https://github.com/brave/brave-browser/issues/10794

**Comment by jumde** [comment]
CI passed here: https://ci.brave.com/job/brave-core-build-pr/job/PR-6046/70/
